### PR TITLE
SEQNG-877 Show the appropriate NIFS fields on the steps table

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds support to show NIFS fields on the steps table

Seqexec - GS-2018B-Q-0-98 2019-03-12 11-07-48

### CODE REVIEW

File: modules/seqexec/model/src/test/scala/seqexec/model/ModelLensesSpec.scala [hunk 1]
```diff
@@ -64,4 +64,12 @@ final class ModelLensesSpec extends CatsSuite with ModelLenses {
            OptionalTests(instrumentDisperserLambdaO))
   checkAll("instrument observing mode Optional",
            OptionalTests(instrumentObservingModeO))
+  checkAll("instrument disperser mode lambda Optional",
+           OptionalTests(instrumentDisperserLambdaO))
+  checkAll("instrument decker Optional",
+           OptionalTests(instrumentDeckerO))
```
Comment (sraaphorst) [inline modules/seqexec/model/src/test/scala/seqexec/model/ModelLensesSpec.scala]: I guess decker and disperser mode were omitted previously?

Comment (cquiroz) [reply modules/seqexec/model/src/test/scala/seqexec/model/ModelLensesSpec.scala]: Yes, I forgot to add tests to those lenses

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/ModelOps.scala [hunk 2]
```diff
@@ -115,6 +120,14 @@ object ModelOps {
     }
   }
 
+  implicit class CamelCaseOps(val s: String) extends AnyVal {
+    def toCamelCase: String =
+      (s.toList match {
+        case Nil       => Nil
+        case x :: rest => x.toUpper :: rest.map(_.toLower)
+      }).mkString
```
Comment (sraaphorst) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/ModelOps.scala]: I'm not 100% clear how this works with camel case. Isn't it just capitalizing the first character and making everything else lowercase?

Comment (cquiroz) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/ModelOps.scala]: Right, maybe this shouldn't be called camelCase. What name would it be more appropriate?

Comment (jluhrs) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/ModelOps.scala]: I was going to say the same, but Sebastian beat me.

Comment (jluhrs) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/ModelOps.scala]: 'PhraseCase' ?

Comment (sraaphorst) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/ModelOps.scala]: I think it's generally called TitleCase.

Comment (cquiroz) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/model/ModelOps.scala]: i  looked around and settled on `Sentence case`. There are multiple versions but it seems `Title Case` will imply more than one capitalized letter and I only wanted the first one capitalized

**Pull request merged**